src/main/java/seedu/mark/storage/JsonAdaptedParagraph.java

```diff
@@ -0,0 +1,86 @@
+package seedu.mark.storage;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import seedu.mark.commons.exceptions.IllegalValueException;
+import seedu.mark.logic.parser.ParserUtil;
+import seedu.mark.logic.parser.exceptions.ParseException;
+import seedu.mark.model.annotation.Annotation;
+import seedu.mark.model.annotation.Paragraph;
+import seedu.mark.model.annotation.ParagraphContent;
+import seedu.mark.model.annotation.ParagraphIdentifier;
+import seedu.mark.model.annotation.PhantomParagraph;
+import seedu.mark.model.annotation.TrueParagraph;
+
+/**
+ * Jackson-friendly version of {@link Paragraph}.
+ */
+public class JsonAdaptedParagraph {
+
+    private final String pid;
+    private final String content;
+    private final JsonAdaptedAnnotation annotation;
+
+    /**
+     * Constructs a {@code JsonAdaptedParagraph} with the given paragraph details.
+     */
+    @JsonCreator
+    public JsonAdaptedParagraph(@JsonProperty("pid") String pid,
+                                @JsonProperty("content") String content,
+                                @JsonProperty("annotation") JsonAdaptedAnnotation annotation) {
+        this.pid = pid;
+        if (content == null) {
```

Comment:
Is it better to require `content` to be non-null and do the check before calling the constructor? (Minor point, can fix or don't-fix in future.)